cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperator.java

```diff
@@ -583,9 +583,9 @@ public void checkCustomCaSecret(CertificateAuthority ca, Secret certSecret, Secr
             if (!reason.isEmpty()) {
                 String reasons = reason.stream().collect(Collectors.joining(", "));
                 Future<Void> zkRollFuture;
-                Predicate<Pod> rollPodAndLogReason = pod -> {
+                Function<Pod, String> rollPodAndLogReason = pod -> {
                     log.debug("{}: Rolling Pod {} to {}", reconciliation, pod.getMetadata().getName(), reasons);
-                    return true;
+                    return "Rolling Pod " + pod.getMetadata().getName() + " to " + reasons;
```

Comment:
I don't think we need to include the "Rolling Pod ... due to " part of this message. We can add the pod id info where the message used for logging (i.e. in the KafkaRoller).
```suggestion
                    return reasons;
```

A similar comment applies to all the other functions.